### PR TITLE
ci: add canaries workflow

### PR DESCRIPTION
**Pull request by iartemiev**
#### Description of changes

Adds a canary workflow to our CircleCI pipeline that will run every hour.
The workflow:
1. Scaffolds new applications with each of our supported frameworks
2. Installs `aws-amplify@latest` from npm
3. Calls the library from the app
4. Builds for dev and starts the app
5. Builds for prod and starts the app

The workflow will fail and notify team members if it encounters errors or times out during any of these steps.

**After merge to main**
* Create a Scheduled Trigger in CircleCI that executes hourly and targets the `canaries` workflow (https://circleci.com/docs/2.0/scheduled-pipelines/#project-settings)

**Note: I'm adding the new orb inline in the config for now in case we need to make changes to any of the job definitions. I will extract it into an actual orb after we've confirmed that the new workflow runs successfully for several weeks.**

#### Description of how you validated changes
Here's the workflow running from a scheduled trigger on this branch: https://app.circleci.com/pipelines/github/aws-amplify/amplify-js?branch=test-getting-started-flow

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1603,3 +1611,286 @@ workflows:
...
+          build-dir: build
+      # Next
+      - getting-started-smoke-test/web:
+          name: Next.js - latest
+          npx-command: create-next-app
+          framework: nextjs
+          main-file-path: pages/_app.js
+          dev-start: dev
+          build-dir: build
+          ssr: true
```
Comment (jamesaucode) [inline .circleci/config.yml:1644]: How is the `ssr` param used? Is it to run `next start` instead of using `serve` ?

Comment (iartemiev) [reply .circleci/config.yml:1644]: Yep, that's exactly it

File: .circleci/config.yml [hunk 2]
```diff
@@ -1603,3 +1608,349 @@ workflows:
                 - release
           requires:
             - deploy
+  # Scheduled smoke test workflow
+  # Jobs are pulled from the getting-started-smoke-test inline orb defined below
+  canaries:
+    when:
+      and:
+        - equal: [scheduled_pipeline, << pipeline.trigger_source >>]
+        # - equal: ['canaries', << pipeline.schedule.name >>]
```
Comment (nickarocho) [inline .circleci/config.yml:1617]: Probably missed this in the description, but is this line still here for later use after merging to main?

File: .circleci/config.yml [hunk 3]
```diff
@@ -1532,15 +1537,15 @@ workflows:
...
-                # - integ_duplicate_packages:
-                #     requires:
-                #       - integ_setup
-                #       - build
-                #     filters:
-                #       <<: *releasable_branches
-                #     matrix:
-                #       parameters:
-                #         <<: *test_browsers
+      # - integ_duplicate_packages:
```
Comment (ashika01) [inline .circleci/config.yml:1540]: curious: what is this artifact?

File: .circleci/config.yml [hunk 4]
```diff
@@ -1603,3 +1608,349 @@ workflows:
...
+      android-executor:
+        machine:
+          image: android:202102-01
+        resource_class: large
+
+    common_env_vars: &common_env_vars
+      environment:
+        IMPORT_STATEMENT: import {Logger} from 'aws-amplify';
+        LIBRARY_STATEMENT: Logger.LOG_LEVEL='DEBUG';
+        MAX_WAIT_ON_TIMEOUT: 180000 # 3 minutes; how long we wait for the build to succeed before failing the job
```
Comment (nickarocho) [inline .circleci/config.yml:1731]: Just curious, how did you land on 3 minutes for the timeout?

**Review by nickarocho — APPROVED** [review]
LGTM! Just left some clarifying questions for my own curiosity.

This is a huuuge improvement we've needed for a long time, great work Ivan!! 🎉

File: .circleci/config.yml (hunk 2 above)
Comment (iartemiev) [reply .circleci/config.yml:1617]: Good catch, I need to uncomment this

File: .circleci/config.yml (hunk 4 above)
Comment (iartemiev) [reply .circleci/config.yml:1731]: I added ~50% to the average successful build time

File: .circleci/config.yml [hunk 5]
```diff
@@ -1603,3 +1608,349 @@ workflows:
...
+        resource_class: large
+
+      android-executor:
+        machine:
+          image: android:202102-01
+        resource_class: large
+
+    common_env_vars: &common_env_vars
+      environment:
+        IMPORT_STATEMENT: import {Logger} from 'aws-amplify';
```
[the remark this answers is not in the document]
Comment (iartemiev) [reply .circleci/config.yml:1729]: This is the [Logger util](https://docs.amplify.aws/lib/utilities/logger/q/platform/js/) in Core. I'm calling it here in order for aws-amplify not to get tree-shaken out of the prod builds. Could have invoked any other function that's exported from `aws-amplify`, but this one requires the least amount of effort 😄

File: .circleci/config.yml [hunk 6]
```diff
@@ -1603,3 +1608,349 @@ workflows:
...
+              label: Pod Install
+              command: npx pod-install
+          - run-with-retry:
+              label: Call Amplify library in code
+              command: |
+                echo "$IMPORT_STATEMENT" | cat - << parameters.main-file-path >> | tee << parameters.main-file-path >>
+                echo "$LIBRARY_STATEMENT" >> << parameters.main-file-path >>
+          - run:
+              background: true
+              command: xcrun simctl boot "iPhone 13" || true
```
[the remark this answers is not in the document]
Comment (iartemiev) [reply .circleci/config.yml]: Updated

File: .circleci/config.yml (hunk 2 above)
Comment (iartemiev) [reply .circleci/config.yml]: Updated

**Pull request merged**